packages/ui/src/components/Radio/radio.tsx

```diff
@@ -0,0 +1,56 @@
+import { Radio as DefaultRadio } from 'antd'
+import React, { useCallback } from 'react'
+
+import { sendAnalyticsCheckEvent, extractChildrenContent } from '../_utils/analytics'
+import { Typography } from '../Typography'
+
+import type { TypographyTextProps } from '../Typography'
+import type { RadioProps as DefaultRadioProps } from 'antd'
+
+const RADIO_CLASS_PREFIX = 'condo-radio'
+
+type CondoRadioProps = {
+    label?: string
+    labelProps?: TypographyTextProps
+}
+
+export type RadioProps = Pick<DefaultRadioProps, 'autoFocus' | 'defaultChecked' | 'disabled' | 'onChange' | 'checked' | 'value' | 'children'>
+& CondoRadioProps
+
+interface IRadio {
+    (props: RadioProps): React.ReactElement
+}
+
+const Radio: IRadio = (props) => {
+    const { label, labelProps, disabled, onChange, children, ...rest } = props
+
+    const handleChange = useCallback((event) => {
+        const stringContent = label ? label : extractChildrenContent(children)
+        if (stringContent) {
+            sendAnalyticsCheckEvent('Radio', { value: stringContent })
+        }
+
+        if (onChange) {
+            onChange(event)
+        }
+    }, [onChange, children, label])
+
+    return (
+        <DefaultRadio
+            {...rest}
+            prefixCls={RADIO_CLASS_PREFIX}
+            disabled={disabled}
+            onChange={handleChange}
+        >
+            {
+                label
+                    ? <Typography.Text size='medium' disabled={disabled} {...labelProps}>{label}</Typography.Text>
```

Comment:
So this radio will render wither label or children?

Reply:
Yup, you can specify label with predefined Typography wrapper or you can do it by your own